### PR TITLE
[BP-62] Add more test for Bookkeeper.

### PR DESCRIPTION
**Pull request by horizonzy**
### Motivation
This is the sixth PR for the batch read(https://github.com/apache/bookkeeper/pull/4051) feature.

Add more test for Bookkeeper.

### CODE REVIEW

**Review by hangc0276 — APPROVED** [review]

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java [hunk 1]
```diff
@@ -610,6 +610,185 @@ public void testReadWriteWithV2WireProtocol() throws Exception {
...
+                try (LedgerHandle lh = bkc.createLedger(2, 2, 2,
+                        digestType, "testPasswd".getBytes())) {
+                    ledgerId = lh.getId();
+                    for (int i = 0; i < numEntries; i++) {
+                        lh.addEntry(data);
+                    }
+                }
+                try (LedgerHandle lh = bkc.openLedger(ledgerId, digestType, "testPasswd".getBytes())) {
+                    assertEquals(numEntries - 1, lh.readLastConfirmed());
+                    //V2 protocol not support batch read. In theory, it will throw UnsupportedOperationException.
```
[the remark this answers is not in the document]
Comment (horizonzy) [reply bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java:632]: Yes, I will correct it.

**Comment by hangc0276** [comment]
@horizonzy  Please take a look at the failed CI
```
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 120.0 s <<< FAILURE! -- in org.apache.bookkeeper.benchmark.TestBenchmark
Error:  org.apache.bookkeeper.benchmark.TestBenchmark.testReadThroughputLatency -- Time elapsed: 120.0 s <<< ERROR!
org.junit.runners.model.TestTimedOutException: test timed out after 120 seconds
	at java.lang.Thread.sleep(Native Method)
	at org.apache.bookkeeper.benchmark.BenchReadThroughputLatency.readLedger(BenchReadThroughputLatency.java:128)
	at org.apache.bookkeeper.benchmark.BenchReadThroughputLatency.main(BenchReadThroughputLatency.java:270)
	at org.apache.bookkeeper.benchmark.TestBenchmark.testReadThroughputLatency(TestBenchmark.java:128)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)

[INFO] Running org.apache.bookkeeper.benchmark.TestBenchmark
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 120.0 s <<< FAILURE! -- in org.apache.bookkeeper.benchmark.TestBenchmark
Error:  org.apache.bookkeeper.benchmark.TestBenchmark.testReadThroughputLatency -- Time elapsed: 120.0 s <<< ERROR!
org.junit.runners.model.TestTimedOutException: test timed out after 120 seconds
	at java.lang.Thread.sleep(Native Method)
	at org.apache.bookkeeper.benchmark.BenchReadThroughputLatency.readLedger(BenchReadThroughputLatency.java:128)
	at org.apache.bookkeeper.benchmark.BenchReadThroughputLatency.main(BenchReadThroughputLatency.java:270)
	at org.apache.bookkeeper.benchmark.TestBenchmark.testReadThroughputLatency(TestBenchmark.java:128)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
```

**Comment by horizonzy** [comment]
rerun failure checks

**Comment by horizonzy** [comment]
rerun failure checks

**Review by merlimat — APPROVED** [review]

**Pull request merged**